dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java

```diff
@@ -533,30 +544,64 @@ public void Test_Protected_Hidden_Secret_And_Values_Returned_Match_Descriptor()
                 //Using a LinkedHashMap we guarantee we keep the original order on which the elements were sent.
                 int index = 0;
                 for (Entry<String, SecretView> secretEntry : secrets.entrySet()) {
-                    final String key = secretEntry.getKey();
-                    final SecretView view = secretEntry.getValue();
-                    final ParamDescriptor originalParam = appDescriptorParamsMap.get(key);
-                    if (view.getSecret().isHidden()) {
-                        Assert.assertEquals(AppsHelper.HIDDEN_SECRET_MASK, new String(view.getSecret().getValue()));
-                    } else {
-                        Assert.assertEquals(originalParam.getString(), new String(view.getSecret().getValue()));
+                    try (final StringWriter writer = new StringWriter()) {
+
+                        final JsonGenerator jsonGenerator = createJsonGenerator(writer);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this JsonGenerator object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)